rust/actyx/ax-core/build.rs

```diff
@@ -1,35 +1,15 @@
-use std::process::Command;
-
-fn git_hash() -> String {
-    // https://stackoverflow.com/a/5737794/576180
-    let out = Command::new("git")
-        .arg("status")
-        .arg("--porcelain")
-        .output()
-        .expect("Error running git status --porcelain");
-
-    let dirty = if out.stdout.is_empty() { "" } else { "_dirty" };
-
-    let git_head_out = Command::new("git")
-        .arg("rev-parse")
-        .arg("HEAD")
-        .output()
-        .expect("Error running git rev-parse --short HEAD");
-
-    let hash = String::from_utf8_lossy(&git_head_out.stdout).trim().to_string();
-
-    format!("{}{}", hash, dirty)
-}
+use std::vec::Vec;
 
+// We need this build script to propagate the target variables through compilation
+// instead of relying on #[cfg(target_arch = "...")] which isn't always "guessable"
+// this way, we don't guess, we use what Cargo uses, we split and propagate.
+// See the following link for more information:
+// https://doc.rust-lang.org/cargo/appendix/glossary.html#target
```

Comment:
To me this doesn’t explain the issue in full: What would be wrong with using `#[cfg(target_os = ... )]` and `#[cfg(target_arch = ...)]` in `version.rs` at this point? The previous need for AX_ARCH stemmed only from wanting to distinguish arm6 from arm7, but we don’t need that anymore. This means `ax_core` wouldn’t have build.rs at all, which I consider a good thing.

Reply:
> This means ax_core wouldn’t have build.rs at all, which I consider a good thing.

I agree that it would be good, but it isn't that simple.

1. The issue with using `#[cfg(target_arch = ...)]` is that it is error prone, it was causing issues in the Android build because the arch wasn't in the list (was falling through to the panic) and I wasn't able to simply check which one it was.
2. The `build.rs` is required because the environment variables from cargo (like `TARGET`) are available during building, _not compilation_. They need to be propagated to `rustc` so they're available later for the versioning string.